src/core/constraints.c

```diff
@@ -994,6 +1000,59 @@ constrain_size_limits (MetaWindow         *window,
 }
 
 static gboolean
+constrain_external_signal (MetaWindow         *window,
+                           ConstraintInfo     *info,
+                           ConstraintPriority priority,
+                           gboolean           check_only)
+{
+  MetaRectangle *start_rect = NULL;
+  gint new_width, new_height;
+  gboolean too_small = FALSE;
+
+
+  if (priority > PRIORITY_GEOMETRY_ALLOCATE_SIGNAL)
+    return TRUE;
+
+  /* Determine whether constraint applies; exit if it doesn't.
+   *
+   * Note: The old code didn't apply this constraint for fullscreen or
+   * maximized windows--but that seems odd to me.  *shrug*
+   */
+  if (info->action_type == ACTION_MOVE)
+    return TRUE;
+
+  window->currently_allocating_rect = info->current;
+  meta_window_start_geometry_allocation (window);
+
+  too_small = !meta_rectangle_could_fit_rect (&info->current,
+                                              &window->currently_allocating_rect);
+
+  if (check_only || !too_small)
+    return too_small;
```

Comment:
Actually, it looks like this check was the wrong way around as well! Surprised it works ...